onnxruntime/python/tools/transformers/benchmark.py

```diff
@@ -333,11 +338,15 @@ def run_pytorch(
             cache_dir=cache_dir,
             custom_model_class=model_class,
         )
-        tokenizer = AutoTokenizer.from_pretrained(model_name, cache_dir=cache_dir)
 
-        max_input_size = (
-            tokenizer.max_model_input_sizes[model_name] if model_name in tokenizer.max_model_input_sizes else 1024
-        )
+        if config.model_type == "vit":
+            max_input_size = 1024 # What to use for ViT?
```

Comment:
max_input_size = None

Reply:
The problem is the main loop won't run if the given sequence_length is larger than max_input_size.

So I think we have a few options:
  Change the loop logic to handle VIT as a special case (ignoring sequence_length for VIT)
  Require sequence_length to be 0 for VIT, then max_input_size of 0 also works.
  Leave as is, where sequence_length can be anything, but does nothing.

What do you prefer?

Reply:
see my comments with example code to update sequence_lengths based on model type.